### PR TITLE
Ui fixes

### LINKED ISSUE #28813: A few issues with new panel layout
* Defaults to "Label" style should be classic

* Player centering completely messed up (Not taking panel size into account)
Auswahl_025
Auswahl_027

* No pixel minimap displayed even if active (Crashes after toggling it onand off with N #28812)
Auswahl_023

* List items: Monsters tab messed up/missing lines. (Items got fixed it seems)
Auswahl_026

* Panel Admin lable artefacts in curses mode after moving some around.
Auswahl_029

* Panel manager hardly usable on minimum resolution
Auswahl_022

### LINKED ISSUE #28890: The Compass panel doesn't show creature names in the Classic sidebar layout.
The Compass panel doesn't show creature names in the Classic sidebar layout.

In the 'Labels' and 'Compact' views the Compass correctly shows the location of each creature, with the creature's name listed below the compass.
compact

However when in the 'Classic' layout only the location letters are displayed, the creature's names are not listed.
classic

**Versions and configuration(please complete the following information):**
OS: Windows 7
Game Version: 0.D-841-g462722b
Graphics version: Tiles

### LINKED ISSUE #29059: Map on side?
**Describe the bug**  
I don't know what its called but the Info Panel on the side is not all the way down the side

**To Reproduce**  
Happens on new Character and Maps 

**Expected behavior**  
The Panel would be all the way down the side

**Screenshots**  
Screenshot (57)

**Versions and configuration(please complete the following information):**
 - OS: Windows 10
 - Game Version: 0.D-1062-g14a0cbf
 - Graphics version: Tiles
 - Mods loaded:
Dark Days Ahead
Medieval and Historic Content
More Survival Tools
Classic Roguelike Classes
StatsThroughSkills
Vehicle Additions Pack
Tanks and Other Vehicles
Boats
Folding Parts pack
Filthy Clothing 
Makeshift Items Mod
Classes and Scenarios Mod
Mutant NPCs
Necromancy
Mythological Replicas
Modular Turrets
Salvaged Robots
Arcana and Magic Item
Disable NPC Needs
More Locations
Fuji's More Buildings
Urban Development
**Additional context**  
Sorry if I can't be of more help as I don't know much about this kinda stuff

### PR DESCRIPTION
**Pull request by KorGgenT**
#### Summary

```SUMMARY: Features "finish UI panel features and fix some visual problems"```
#### Purpose of change

the feature was incomplete. i wanted to fix the rest of them.
Closes #28813
Fixes #28890
Fixes #29059
#### Describe the solution

- Center view based on sidebar
- fix border on monster list view
- add option to add spacers between panels
- log expands to unused space
- fix Perception display string in classic mode

### CODE REVIEW

File: src/game.cpp [hunk 1]
```diff
@@ -3573,9 +3573,22 @@ void game::draw_panels( size_t column, size_t index )
     auto &mgr = panel_manager::get_manager();
     int y = 0;
     const bool sidebar_right = get_option<std::string>( "SIDEBAR_POSITION" ) == "right";
+    int spacer;
```
Comment (AMurkin) [inline src/game.cpp]:
```suggestion
    int spacer = get_option<bool>( "SIDEBAR_SPACERS" ) ? 1 : 0;
```

File: src/game.cpp [hunk 2]
```diff
@@ -3573,9 +3573,22 @@ void game::draw_panels( size_t column, size_t index )
...
+    get_option<bool>( "SIDEBAR_SPACERS" ) ? spacer = 1 : spacer = 0;
```
Comment (AMurkin) [inline src/game.cpp]:
```suggestion
```

File: src/game.cpp [hunk 3]
```diff
@@ -7894,10 +7908,11 @@ game::vmenu_ret game::list_monsters( const std::vector<Creature *> &monster_list
...
 
-    mvwputch( w_monsters_border, getmaxy( w_monsters ) - 2, 0, BORDER_COLOR, LINE_XOXO ); // |
-    mvwputch( w_monsters_border, getmaxy( w_monsters ) - 2, width - 1, BORDER_COLOR, LINE_XOXO ); // |
-    mvwputch( w_monsters_border, getmaxy( w_monsters ) - 1, 0, BORDER_COLOR, LINE_XOXO ); // |
-    mvwputch( w_monsters_border, getmaxy( w_monsters ) - 1, width - 1, BORDER_COLOR, LINE_XOXO ); // |
+    for( int i = 1; i < getmaxy( w_monsters ) - 1; i++ ) {
+        mvwputch( w_monsters_border, i, 0, BORDER_COLOR, LINE_XOXO ); // |
+        mvwputch( w_monsters_border, i, width - 1, BORDER_COLOR, LINE_XOXO ); // |
+    }
+
```
Comment (AMurkin) [inline src/game.cpp]:
```suggestion
```

File: src/panels.cpp [hunk 4]
```diff
@@ -130,7 +130,7 @@ std::pair<nc_color, std::string> per_string( const player &p )
     } else if( p.get_per() < p.get_per_base() ) {
         clr = c_red;
     }
-    return std::make_pair( clr, "Int " + ( p.get_per() < 100 ? to_string( p.get_per() ) : "++" ) );
+    return std::make_pair( clr, "Per " + ( p.get_per() < 100 ? to_string( p.get_per() ) : "++" ) );
```
Comment (AMurkin) [inline src/panels.cpp]: These labels should be translated.

**Comment by ZhilkinSerg** [comment]
View center is in the center of the screen, not on the player.

image

**Comment by KorGgenT** [comment]
so far i don't know what part of the code draws that cursor. i only hope i can find it soon.

**Comment by OzoneH3** [comment]
Item/Monster Tab problems:


Curses:
* Item/Monster Tab Trail drawing not working correctly
* View still not centered.
* Rain is clearing the @ character symbol.



Auswahl_032

**Comment by tinukedaya** [comment]
If you can, please also check the #28890 #28914 #28915 and #28916

**Comment by kevingranade** [comment]
This pull request has been mentioned on **Cataclysm: Dark Days Ahead**. There might be relevant details there:

https://discourse.cataclysmdda.org/t/yall-messed-up-the-interface/19310/8

**Comment by anonym** [comment]
@KorGgenT, let me first say I already love the new possibilities with the panel manager! Great work! I was appalled to see the ignorant reactions on reddit etc, so I thought you might want to hear that some of understand where this is going and appreciate your contribution. :+1:

So I've tested this branch since I was really looking forward to commit f22d2ee, but noticed that it introduces a regression: whenever I perform and action, the "Look around" square is drawn for a short time around the absolute middle square (i.e. where the character would be without this commit). Looking at the code one wouldn't expect this, but I assume there are more places where the code makes assumptions about the "center" that isn't true any more.

I wonder, is it intentional that the game view (i.e. the tiles) extends "under" the side bar? Perhaps you want to allow players to make minimal panels and thus see more? I was just thinking that if we limited the game view so it stopped where the side bar begins (when it is shown) then those assumptions about the "center" I talked about would become true again and you can stop playing whack-a-mole adding correcting offsets all over the code base like f22d2ee. Of course, I haven't looked into any of this in detail, so I might be completely off.

**Comment by kevingranade** [comment]
IDK if it happens on tiles yet, but on curses the rain animation causes most of the map to shoft to one side by a large number of characters.
Mostly before it happens:
weather-ui-shift-before
In an intermediate state:
weather-ui-shift-during
After it hash shifted the whole map:
weather-ui-shift-after

**Comment by kevingranade** [comment]
This pull request has been mentioned on **Cataclysm: Dark Days Ahead**. There might be relevant details there:

https://discourse.cataclysmdda.org/t/interface-issues/19394/2

**Comment by ZhilkinSerg** [comment]
> so far i don't know what part of the code draws that cursor. i only hope i can find it soon.

This code draws this cursor:

https://github.com/CleverRaven/Cataclysm-DDA/blob/2bf28eab3586d0209e4665bc84ef4dc295408f20/src/cata_tiles.cpp#L1224-L1229

**Comment by ZhilkinSerg** [comment]
Cursor now appears on player character when moving:

image

**Comment by OzoneH3** [comment]
- [x] Player centering in curses works now.

Had to add ```#if (defined TILES || defined _WIN32 || defined WINDOWS)``` to
```
void to_map_font_dimension( int &w, int &h );
void from_map_font_dimension( int &w, int &h );
void to_overmap_font_dimension( int &w, int &h );
void reinitialize_framebuffer();
```
functions in game.cpp to make it compile.

List items:
The 'X' from ```draw_trail_to_square``` is to the right of the item:


Rain dose not artifact anymore but is not raining on the left side by the amount of the offset:


List monsters:
Cursor is off by one when switching to 'x look around' on monster.

**Comment by KorGgenT** [comment]
> * [x]  Player centering in curses works now.
> 
> Had to add `#if (defined TILES || defined _WIN32 || defined WINDOWS)` to
> 
> ```
> void to_map_font_dimension( int &w, int &h );
> void from_map_font_dimension( int &w, int &h );
> void to_overmap_font_dimension( int &w, int &h );
> void reinitialize_framebuffer();
> ```
> functions in game.cpp to make it compile.

i was hoping the #if was going to do just that in sdltiles.cpp. you were having issues compiling it? i didn't particularly want to add the #if to every line in game.cpp those functions were defined; do you have any thoughts as to the correct way to get this to work with the header file?

**Comment by OzoneH3** [comment]
The issues come from game.cpp.
On the linux terminal version sdltiles dose not get included.
Gorgon has the same problem.

**Comment by KorGgenT** [comment]
then do i need to rename the header file? if it's not including even if i'm telling it to, no wonder i'm having problems

**Comment by OzoneH3** [comment]
You will have to put the prototypes and non tiles functions which return nothing somewhere else, where both console and tiles can find them (not too sure where best through).

**Comment by kevingranade** [comment]
Honestly this can wait if it checks out otherwise, but I see bullet trails from NPCs shooting at things being drawn with an offset.  Somewhere in animation.cpp perhaps.

**Comment by ZhilkinSerg** [comment]
> 
> 
> Honestly this can wait if it checks out otherwise, but I see bullet trails from NPCs shooting at things being drawn with an offset. Somewhere in animation.cpp perhaps.
> Also SCT and hit animations.

Probably the same issue that was affecting snow/rain animation?

**Comment by KorGgenT** [comment]
jenkins rebuild

**Comment by kevingranade** [comment]
Bullet trails working, SCT messages still offset.
I don't think that's important enough to block this, I'll open an issue if I merge this before it's fixed.

**Comment by kevingranade** [comment]
Is it intentional that in compact view the minimap does not draw a border? It has one, but it looks like it's blanks.

**Comment by Tharn** [comment]
The issue with the 'view' cursor being drawn when it shouldn't is still there, albeit difficult to reproduce. I made the view cursor appear next to my character by switching windows and coming back to CDDA. I may have clicked on the screen as well.

**Comment by gc2020x4** [comment]
@kevingranade @KorGgenT 
Kevin Grenade must leave the project forever.

**Pull request merged**